pkg/controllers/core/catalog_controller_test.go

```diff
@@ -539,9 +540,70 @@ func TestCatalogdControllerReconcile(t *testing.T) {
 			} else {
 				assert.Error(t, err)
 			}
-
+			tt.catalog.Status.LastPollAttempt = metav1.Time{}
 			diff := cmp.Diff(tt.expectedCatalog, tt.catalog, cmpopts.IgnoreFields(metav1.Condition{}, "Message", "LastTransitionTime"))
 			assert.Empty(t, diff, "comparing the expected Catalog")
 		})
 	}
 }
+
+func TestPolling(t *testing.T) {
+	for name, tc := range map[string]struct {
+		catalog              *catalogdv1alpha1.Catalog
+		expectedRequeueAfter time.Duration
+	}{
+		"Catalog with tag based image ref without any poll interval specified, requeueAfter set to 0, ie polling disabled": {
+			catalog: &catalogdv1alpha1.Catalog{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       "test-catalog",
+					Finalizers: []string{fbcDeletionFinalizer},
+				},
+				Spec: catalogdv1alpha1.CatalogSpec{
+					Source: catalogdv1alpha1.CatalogSource{
+						Type: "image",
+						Image: &catalogdv1alpha1.ImageSource{
+							Ref: "someimage:latest",
+						},
+					},
+				},
+			},
+			expectedRequeueAfter: time.Second * 0,
+		},
+		"Catalog with tag based image ref with poll interval specified, requeueAfter set to wait.jitter(pollInterval)": {
+			catalog: &catalogdv1alpha1.Catalog{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       "test-catalog",
+					Finalizers: []string{fbcDeletionFinalizer},
+				},
+				Spec: catalogdv1alpha1.CatalogSpec{
+					Source: catalogdv1alpha1.CatalogSource{
+						Type: "image",
+						Image: &catalogdv1alpha1.ImageSource{
+							Ref:          "someimage:latest",
+							PollInterval: &metav1.Duration{Duration: time.Minute * 5},
+						},
+					},
+				},
+			},
+			expectedRequeueAfter: time.Minute * 5,
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			reconciler := &CatalogReconciler{
+				Client: nil,
+				Unpacker: source.NewUnpacker(
+					map[catalogdv1alpha1.SourceType]source.Unpacker{
+						catalogdv1alpha1.SourceTypeImage: &MockSource{result: &source.Result{
+							State: source.StateUnpacked,
+							FS:    &fstest.MapFS{},
+						}},
+					},
+				),
+				Storage: &MockStore{},
+			}
+			res, _ := reconciler.reconcile(context.Background(), tc.catalog)
+			assert.GreaterOrEqual(t, res.RequeueAfter, tc.expectedRequeueAfter)
+			assert.LessOrEqual(t, res.RequeueAfter, wait.Jitter(res.RequeueAfter, requeueJitterMaxFactor))
```

Comment:
I think as written this assertion is irrespective of the test case and only operates on `wait.Jitter()`.

Reply:
This comment is still valid - I don't think this is testing what you mean for it to test.

Reply:
+1, looks like this should be:
```suggestion
			assert.LessOrEqual(t, res.RequeueAfter, wait.Jitter(tc.expectedRequeueAfter, requeueJitterMaxFactor))
```

Reply:
Sorry missed this, fixed. PTAL